cmsdev/internal/lib/k8s/k8s.go

```diff
@@ -354,6 +354,30 @@ func GetNodeNames(params ...string) ([]string, error) {
 	return names, err
 }
 
+// Given a namespace and name, returns the matching configmap
+func GetConfigMap(namespace, name string) (cm coreV1.ConfigMap, err error) {
+	common.Debugf("Retrieving Kubernetes ConfigMap in namespace %s with name %s", namespace, name)
+
+	clientset, err := GetClientset()
+	if err != nil {
+		return
+	}
+
+	cmlist, err := clientset.CoreV1().ConfigMaps(namespace).List(v1.ListOptions{})
+	if err != nil {
+		return
+	}
+
+	for _, cm = range cmlist.Items {
+		if cm.ObjectMeta.Name == name {
+			common.Debugf("Found Kubernetes ConfigMap in namespace %s with name %s", namespace, name)
+			return
```

Comment:
Is this function just checking for the existence of the cm? Or, is it supposed to return the actual cm data? Am I missing something? I'm not a go person, by do you need to `return cm`?

Reply:
It's a golang thing -- this function does return **cm** when it finds it. 

In go you can name the return variables in the function statement. In this case, in the function statement I declared that the cm variable would hold the return value for the ConfigMap (and the err variable would hold the error value, if any). That's why all of the return statements have no arguments.

Reply:
Got it. Thanks.

Reply:
Note that this type of syntax is optional -- you can declare a function and not name the return variables (you do still have to declare their types), and then use return statements in the more traditional way. It all comes down to what makes more sense for the given function.